### DISCUSSION TITLE
How to remove .html extension from page urls (gh-pages)?

### DISCUSSION THREAD

**Discussion by MarkParker5** [squidfunk/mkdocs-material#6046] [Q&A] [answered]
How to remove .html extension from links? I published docs https://stark.markparker.me (repo https://github.com/MarkParker5/STARK) using gh pages and all my links have .html at the end. If I type the url without .html, it still works, but `a` tags have .html in `href`.

**Comment by alexvoss** [comment]
I don't have experience with gh-pages but the compilation of the pages is controlled by [the `use_directory_urls` option in MkDocs](https://www.mkdocs.org/user-guide/configuration/#use_directory_urls). When I build your documentation it produced HTML files instead of directories with `index.html` files in them. However, I cannot see you using `use_directory_urls` in your `mkdocs.yml`. Odd, the defailt is `true` according to the MkDocs documentation and that works for me with my sites.

**Reply by alexvoss** [reply, marked as answer]
Ok, you are using the [Offline Plugin](https://squidfunk.github.io/mkdocs-material/plugins/offline/), which explains the behavior: "Additionally, the plugin automatically disables the [use_directory_urls](https://www.mkdocs.org/user-guide/configuration/#use_directory_urls) setting, ensuring that users can open your documentation directly from the local file system." As far as I can see, you can either have offline use or URLs without the `.html` extension.

**Reply by squidfunk** [reply]
Correct! The [offline plugin](https://squidfunk.github.io/mkdocs-material/plugins/offline/) is intended for distributing your documentation as a downloadable `.zip` file, which you can view from the local file system without the use of a server. For this to work, the `.html` extension is mandatory. You should not use the plugin when you deploy your documentation.

**Reply by alexvoss** [reply]
You should be able to enable/disable the plugin using an environment variable as [described here](https://squidfunk.github.io/mkdocs-material/plugins/offline/#config.enabled).

**Reply by MarkParker5** [reply]
Yes that's it! Thank you!
